hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/model/ArtifactUpload.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (c) 2015 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.repository.model;
+
+import java.io.InputStream;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+
+import org.eclipse.hawkbit.repository.RegexCharacterCollection;
+import org.eclipse.hawkbit.repository.RegexCharacterCollection.RegexChar;
+import org.hibernate.validator.constraints.SafeHtml;
+
+/**
+ * Use to create a new artifact.
+ *
+ */
+public class ArtifactUpload {
+
+    public static final RegexCharacterCollection ILLEGAL_FILENAME_CHARACTERS = new RegexCharacterCollection(
+            RegexChar.GREATER_THAN, RegexChar.LESS_THAN, RegexChar.SLASHES);
+
+    @NotNull
+    private final InputStream inputStream;
+
+    private final long moduleId;
+
+    @NotEmpty
+    @SafeHtml(message = "The filename might contain unsafe HTML code")
```

Comment:
What about `@SafeHtml(WhiteListType.NONE)` instead of handcrafted regex matching? Is it too much for a filename?

Reply:
Another idea is to introduce a default method in `ArtifactManagement` e.g. validateFilename(...) this perfectly fits to all other validations we currently doing and it's a central place

Reply:
The idea was, that in the future this @SafeHtml will be replaced by a custom javax.validation.constraints (e.g. @IllegalCharacters), that takes a collection of RegexChars as argument. This constraint can then be used for every field of an entity that needs to be validated. That way we can use the RegexChars defined for a field both in the constraint and for manual validation in the UI.